lib/src/ruby/module.cc

```diff
@@ -696,15 +692,16 @@ namespace facter { namespace ruby {
 
         // Unfortunately we have to call to_sym rather than using ID2SYM, which is Ruby version dependent
         uint32_t timeout = 0;
-        volatile VALUE timeout_option = ruby.rb_hash_lookup(argv[1], ruby.rb_funcall(ruby.utf8_value("timeout"), ruby.rb_intern("to_sym"), 0));
+        volatile VALUE timeout_option = ruby.rb_hash_lookup(argv[1], ruby.to_symbol("timeout"));
         if (ruby.is_fixednum(timeout_option)) {
             timeout = static_cast<uint32_t>(ruby.rb_num2ulong(timeout_option));
         }
 
-        // Get the on_fail option
+        // Get the on_fail option (defaults to :raise)
         bool raise = false;
```

Comment:
Third argument to `rb_hash_lookup2`, which returns the :raise symbol if not present in the options hash.